src/qiskit_qasm3_import/expression.py

```diff
@@ -286,21 +317,27 @@ def visit_IndexedIdentifier(self, node: ast.IndexedIdentifier):
 
 
 def resolve_condition(
-    node: ast.Expression, symbols: Mapping[str, Symbol]
+    node: ast.Expression, context=None  # symbols: Mapping[str, Symbol]
 ) -> Union[Tuple[Clbit, bool], Tuple[Iterable[Clbit], int]]:
     """A resolver for conditions that can be converted into Qiskit's very basic equality form
     of either ``Clbit == bool`` or ``ClassicalRegister == int``.
 
     This effectively just handles very special outer cases, then delegates the rest of the work to a
     :class:`.ValueResolver`."""
 
+    if isinstance(context, dict):
+        symbols = context
+        context = None
+    else:
+        symbols = context.symbol_table
+
```

Comment:
Similar to before: I don't think this should be optional, and if we're passing the whole thing, we don't need to separate `symbols` from the `State`.

Reply:
I did this early, did not want to break any possible existing calls. Of course in the end, it's not needed.